src/setup.sh

```diff
@@ -294,19 +336,32 @@ function check_prerequisites {
 # Installs all the dependencies for docker-osx-dev.
 #
 function install_dependencies {
-  log_info "Updating HomeBrew"
-  brew update
+  if is_brew_installed ; then
+
+    log_info "Updating HomeBrew"
+    brew update
+
+    brew_install "caskroom/cask/brew-cask" "Cask" "" false
+    brew_install "virtualbox" "VirtualBox" "VBoxManage" true
+    brew_install "boot2docker" "Boot2Docker" "boot2docker" false
+    brew_install "docker-compose" "Docker Compose" "docker-compose" false
+    brew_install "fswatch" "fswatch" "fswatch" false
+    brew_install "coreutils" "GNU core utilities" "greadlink" false
 
-  brew_install "caskroom/cask/brew-cask" "Cask" "" false
-  brew_install "virtualbox" "VirtualBox" "VBoxManage" true
-  brew_install "boot2docker" "Boot2Docker" "boot2docker" false
-  brew_install "docker-compose" "Docker Compose" "docker-compose" false
-  brew_install "fswatch" "fswatch" "fswatch" false
-  brew_install "coreutils" "GNU core utilities" "greadlink" false
+  elif is_port_installed ; then
+
+    log_info "Updating MacPorts"
+    sudo port selfupdate
+
+    port_install "virtualbox" "VirtualBox" "VBoxManage"
```

Comment:
I'm a little concerned that most of the HomeBrew list is duplicated here, but not all. It makes it trickier to maintain, as you have to update two places (albeit, at least here they are close together), and it's not obvious why some of the items are in one list but not another. Although bash is terrible at data structures, I'm wondering if there is a way to create one to make the solution look more obvious. For example, in JavaScript, I might have used an object like this:

```js
var dependencies = [
  {packageName: "virtualBox", readableName: "Virtual Box", commandName: "VBoxManage", brew: false, brewCask: true, ports: true},
  {packageName: "docker-compose", readableName: "Docker Compose", commandName: "docker-compose", brew: true, brewCask: false, ports: false},
  // ...
]
```